addons/l10n_sa_edi_pos/static/src/overrides/components/payment_screen/payment_screen.js

```diff
@@ -0,0 +1,34 @@
+import { PaymentScreen } from "@point_of_sale/app/screens/payment_screen/payment_screen";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { _t } from "@web/core/l10n/translation";
+import { patch } from "@web/core/utils/patch";
+import { markup } from "@odoo/owl";
+
+patch(PaymentScreen.prototype, {
+    //@Override
+    async _finalizeValidation() {
+        await super._finalizeValidation(...arguments);
+        const order = this.currentOrder;
+        // note: isSACompany guarantees order.is_to_invoice()
+        if (order.isSACompany && order.finalized && order.l10n_sa_invoice_edi_state !== "sent") {
+            const orderError = _t("%s by going to Backend > Orders > Invoice", order.pos_reference);
+            const link = markup(
+                `<a target="_blank" href="/odoo/customer-invoices/${
+                    this.currentOrder.raw.account_move
+                }" class="text-info fw-bolder">${_t("Invoice")}</a>`
+            );
```

Comment:
Honestly, I'm not a big fan of the indentation but that's not why I was asked here :smile: Security wise, the markup is all good; No need for escapes and as close to the source as possible :+1:

If you were to change the indentation however, I think something like this looks a lot better but that's for you to decide :wink:

```suggestion
            const href = `/odoo/customer-invoices/${this.currentOrder?.raw?.account_move}`
            const link = markup(
                `<a target="_blank" href=${href} class="text-info fw-bolder">${_t("Invoice")}</a>`
            );
```

Reply:
Thank you for the feedback, done